src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.ItemFactoryWrapper.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Build.Collections;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Evaluation
+{
+    internal partial class LazyItemEvaluator<P, I, M, D>
+    {
+        class ItemFactoryWrapper : IItemFactory<I, I>
```

Comment:
What is this wrapper used for? It seems to do exactly what the wrapped factory is doing, but storing the current item element twice.